numpy/_core/src/multiarray/stringdtype/casts.c

```diff
@@ -1394,6 +1406,143 @@ static PyType_Slot dt2s_slots[] = {
 
 static char *dt2s_name = "cast_Datetime_to_StringDType";
 
+// string to timedelta
+
+static int
+string_to_timedelta(PyArrayMethod_Context *context, char *const data[],
+                                       npy_intp const dimensions[], npy_intp const strides[],
+                   NpyAuxData *NPY_UNUSED(auxdata))
+{
+    PyArray_StringDTypeObject *descr = (PyArray_StringDTypeObject *)context->descriptors[0];
+    npy_string_allocator *allocator = NpyString_acquire_allocator(descr);
+    int has_null = descr->na_object != NULL;
+    int has_string_na = descr->has_string_na;
+    const npy_static_string *default_string = &descr->default_string;
+
+    npy_intp N = dimensions[0];
+    char *in = data[0];
+    npy_timedelta *out = (npy_timedelta *)data[1];
+
+    npy_intp in_stride = strides[0];
+    npy_intp out_stride = strides[1] / sizeof(npy_timedelta);
+
+    while (N--) {
+        const npy_packed_static_string *ps = (npy_packed_static_string *)in;
+        npy_static_string s = {0, NULL};
+        int is_null = NpyString_load(allocator, ps, &s);
+        if (is_null == -1) {
+            PyErr_SetString(
+                    PyExc_MemoryError,
+                    "Failed to load string in string to datetime cast");
+            goto fail;
+        }
+        if (is_null || is_nat_string(&s)) {
```

Comment:
It does not seem logical that if the string is not missing itself, but equal to `NaT`, that then the outcome depends on the `StringDType` - I think it should always be `NPY_DATETIME_NAT`. Maybe easiest to just separate out the behaviour for `is_null` and `is_nat_string`, deal with `is_null` first and then check `is_nat_string` (so that if the default string is `NaT`, that get's caught appropriately)